src/command/utils.rs

```diff
@@ -8,8 +8,34 @@ use walkdir::WalkDir;
 
 /// If an explicit path is given, then use it, otherwise assume the current
 /// directory is the crate path.
-pub fn set_crate_path(path: Option<PathBuf>) -> Result<PathBuf, failure::Error> {
-    Ok(path.unwrap_or_else(|| PathBuf::from(".")))
+pub fn get_crate_path(path: Option<PathBuf>) -> Result<PathBuf, failure::Error> {
+    match path {
+        Some(p) => Ok(p),
+        None => find_manifest_from_cwd(),
+    }
+}
+
+/// Search up the path for the manifest file from the current working directory
+/// If we don't find the manifest file then return back the current working directory
+/// to provide the appropriate error
+fn find_manifest_from_cwd() -> Result<PathBuf, failure::Error> {
+    let cwd = PathBuf::from(".");
+    let parent_path = cwd.canonicalize()?;
+    let mut parent_path = parent_path.as_path();
```

Comment:
This call should be unnecessary, and I think we can wholesale remove this line. Since `PathBuf` derefs to `Path` the `.join` call below should Just Work(tm).

Reply:
Ah yes I see now - after more digging due to conflicting type errors while trying to address this I actually found the pop method on PathBuf... which is exactly what I was doing but better!